lib/linter/linter.js

```diff
@@ -1643,29 +1643,31 @@ class Linter {
         const options = normalizeVerifyOptions(providedOptions, config);
         const languageOptions = config.languageOptions;
 
-        languageOptions.ecmaVersion = normalizeEcmaVersionForLanguageOptions(
-            languageOptions.ecmaVersion
-        );
+        if (config.language === jslang) {
+            languageOptions.ecmaVersion = normalizeEcmaVersionForLanguageOptions(
+                languageOptions.ecmaVersion
+            );
 
-        // double check that there is a parser to avoid mysterious error messages
-        if (!languageOptions.parser) {
-            throw new TypeError(`No parser specified for ${options.filename}`);
-        }
+            // double check that there is a parser to avoid mysterious error messages
+            if (!languageOptions.parser) {
+                throw new TypeError(`No parser specified for ${options.filename}`);
+            }
```

Comment:
This check seems no longer necessary.

Reply:
Yeah, this check was added in one of the first iterations of the flat config implementation and seems to have been unnecessary for a long time. I removed it in https://github.com/eslint/eslint/pull/19003/commits/b20d2bb8aa6ecc48088208540094dd4740ec4331.